src/Features/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensHelpers.cs

```diff
@@ -120,9 +120,77 @@ internal class SemanticTokensHelpers
             var classifiedSpans = await Classifier.GetClassifiedSpansAsync(document, textSpan, cancellationToken).ConfigureAwait(false);
             Contract.ThrowIfNull(classifiedSpans, "classifiedSpans is null");
 
-            // TO-DO: We should implement support for streaming once this LSP bug is fixed:
-            // https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1132601
-            return ComputeTokens(text.Lines, classifiedSpans.ToArray(), tokenTypesToIndex);
+            // Multi-line tokens are not supported by VS (tracked by https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1265495).
+            // Roslyn's classifier however can return multi-line classified spans, so we must break these up into single-line spans.
+            var updatedClassifiedSpans = ConvertMultiLineToSingleLineSpans(text, classifiedSpans.ToArray());
+
+            // TO-DO: We should implement support for streaming if LSP adds support for it:
+            // https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1276300
+            return ComputeTokens(text.Lines, updatedClassifiedSpans, tokenTypesToIndex);
+        }
+
+        private static ClassifiedSpan[] ConvertMultiLineToSingleLineSpans(SourceText text, ClassifiedSpan[] classifiedSpans)
+        {
+            using var _ = ArrayBuilder<ClassifiedSpan>.GetInstance(out var updatedClassifiedSpans);
+
+            for (var spanIndex = 0; spanIndex < classifiedSpans.Length; spanIndex++)
+            {
+                var span = classifiedSpans[spanIndex];
+                text.GetLinesAndOffsets(span.TextSpan, out var startLine, out var startOffset, out var endLine, out var endOffSet);
+
+                // If the start and end of the classified span are not on the same line, we're dealing with a multi-line span.
+                // Since VS doesn't support multi-line spans/tokens, we need to break the span up into single-line spans.
+                if (startLine != endLine)
+                {
+                    var numLinesInSpan = endLine - startLine + 1;
+                    Contract.ThrowIfTrue(numLinesInSpan < 1);
+
+                    for (var currentLine = 1; currentLine <= numLinesInSpan; currentLine++)
+                    {
+                        TextSpan? textSpan;
+
+                        // Case 1: First line of span
+                        if (currentLine == 1)
```

Comment:
beign 1-based is so very very strange.  why?

Reply:
I thought being 1-based would make the variable `numLinesInSpan` easier to understand, but if you think it's more confusing than I can definitely switch it to 0-based.

Reply:
Whoops, you're totally right here - updated!